plugins/inputs/cloudwatch/cloudwatch.go

```diff
@@ -1,67 +1,98 @@
 type (
+	// CloudWatch contains the configuration and cache for the cloudwatch plugin.
 	CloudWatch struct {
-		Region      string `toml:"region"`
-		AccessKey   string `toml:"access_key"`
-		SecretKey   string `toml:"secret_key"`
-		RoleARN     string `toml:"role_arn"`
-		Profile     string `toml:"profile"`
-		Filename    string `toml:"shared_credential_file"`
-		Token       string `toml:"token"`
-		EndpointURL string `toml:"endpoint_url"`
-
-		Period      internal.Duration `toml:"period"`
-		Delay       internal.Duration `toml:"delay"`
-		Namespace   string            `toml:"namespace"`
-		Metrics     []*Metric         `toml:"metrics"`
-		CacheTTL    internal.Duration `toml:"cache_ttl"`
-		RateLimit   int               `toml:"ratelimit"`
-		client      cloudwatchClient
-		metricCache *MetricCache
-		windowStart time.Time
-		windowEnd   time.Time
+		Region           string   `toml:"region"`
+		AccessKey        string   `toml:"access_key"`
+		SecretKey        string   `toml:"secret_key"`
+		RoleARN          string   `toml:"role_arn"`
+		Profile          string   `toml:"profile"`
+		CredentialPath   string   `toml:"shared_credential_file"`
+		Token            string   `toml:"token"`
+		EndpointURL      string   `toml:"endpoint_url"`
+		StatisticExclude []string `toml:"statistic_exclude"`
+		StatisticInclude []string `toml:"statistic_include"`
+
+		Period    internal.Duration `toml:"period"`
+		Delay     internal.Duration `toml:"delay"`
+		Namespace string            `toml:"namespace"`
+		Metrics   []*Metric         `toml:"metrics"`
+		CacheTTL  internal.Duration `toml:"cache_ttl"`
+		RateLimit int               `toml:"ratelimit"`
+
+		client          cloudwatchClient
+		statFilter      filter.Filter
+		metricCache     *metricCache
+		queryCache      *queryCache
+		filteredCache   *filteredCache
+		queryDimensions map[string]*[]dimension
+		windowStart     time.Time
+		windowEnd       time.Time
 	}
 
+	// Metric defines a simplified Cloudwatch metric.
 	Metric struct {
-		MetricNames []string     `toml:"names"`
-		Dimensions  []*Dimension `toml:"dimensions"`
+		StatisticExclude *[]string    `toml:"statistic_exclude"`
+		StatisticInclude *[]string    `toml:"statistic_include"`
+		MetricNames      []string     `toml:"names"`
+		Dimensions       []*Dimension `toml:"dimensions"`
 	}
 
+	// Dimension defines a simplified Cloudwatch dimension (provides metric filtering).
 	Dimension struct {
 		Name  string `toml:"name"`
 		Value string `toml:"value"`
 	}
 
-	MetricCache struct {
-		TTL     time.Duration
-		Fetched time.Time
-		Metrics []*cloudwatch.Metric
+	// metricCache caches automatically fetched metrics.
+	metricCache struct {
+		ttl     time.Duration
+		fetched time.Time
+		metrics []*cloudwatch.Metric
+	}
+
+	// queryCache caches generated queries.
+	queryCache struct {
+		ttl     time.Duration
+		built   time.Time
+		queries []*cloudwatch.MetricDataQuery
+	}
+
+	// filteredCache caches defined metrics and their filters.
+	filteredCache struct {
```

Comment:
Should we combine all these caches into a single cache since they all have the same lifetime?  Do we need both filteredMetric and metricCache, as they both seem to hold the `*cloudwatch.Metric`?